hive-metastore/src/test/java/org/apache/iceberg/hive/HiveTableTest.java

```diff
@@ -325,4 +329,71 @@ public void testNonDefaultDatabaseLocation() throws IOException, TException {
     metastoreClient.dropDatabase(NON_DEFAULT_DATABASE, true, true, true);
   }
 
+  @Test
+  public void testEngineHiveEnabled() throws TException {
+    // Default already created
+    org.apache.hadoop.hive.metastore.api.Table hmsTable = metastoreClient.getTable(DB_NAME, TABLE_NAME);
+
+    assertHiveEnabled(hmsTable, false);
+
+    catalog.dropTable(TABLE_IDENTIFIER);
+
+    // Enable by hive-conf
```

Comment:
It looks like a lot of test cases are mixed together in this method, but separated by comments. I think it would be better to make these separate test methods because then they are run independently, so you can see all of the failures at once, instead of just one failure per run that potentially hides the others.

Reply:
Done (kept true/false tests in a single test case, but separated on config source)